### PR TITLE
feat(chromium): switch to headless=new by default

### PR DESCRIPTION
**Pull request by dgozman**
~~This PR switches to headless=new by default to assess the impact. `PLAYWRIGHT_CHROMIUM_USE_HEADLESS_OLD` environment variable reverses to the old behavior.~~

This PR switches to the `headless=new` by default to assess the impact. To reverse to the old behaviour we recommend `channel: 'chromium-headless-shell'`.

https://github.com/microsoft/playwright/issues/33144

### CODE REVIEW

File: packages/playwright-core/src/server/chromium/chromium.ts [hunk 1]
```diff
@@ -307,10 +307,10 @@ export class Chromium extends BrowserType {
     if (options.devtools)
       chromeArguments.push('--auto-open-devtools-for-tabs');
     if (options.headless) {
-      if (process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW)
-        chromeArguments.push('--headless=new');
-      else
+      if (process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_OLD)
```
Comment (mxschmitt) [inline packages/playwright-core/src/server/chromium/chromium.ts:310]: While passing `--headless=new` to Headless Shell works, and they probably treat it as old, I'm thinking if we should pass `--headless=old` there instead.

**Review by mxschmitt — APPROVED** [review]

File: tests/library/browsercontext-credentials.spec.ts [hunk 2]
```diff
@@ -18,8 +18,8 @@
 import { browserTest as base, expect } from '../config/browserTest';
 
 const it = base.extend<{ isChromiumHeadedLike: boolean }>({
-  isChromiumHeadedLike: async ({ browserName, headless }, use) => {
-    const isChromiumHeadedLike = browserName === 'chromium' && (!headless || !!process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW);
+  isChromiumHeadedLike: async ({ browserName, headless, channel }, use) => {
```
Comment (yury-s) [inline tests/library/browsercontext-credentials.spec.ts:21]: !headless || (browserName === 'chromium' && headless && !process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_OLD && channel !== 'chromium-headless-shell') ?

**Review by yury-s — APPROVED** [review]

**Comment by pavelfeldman** [comment]
Let's do this without the environment variable. Those that are affected by headless=new and are not willing to switch to the headless shell should stay on 1.48 while we are figuring out how to help them.

File: packages/playwright-core/src/server/bidi/bidiChromium.ts [hunk 3]
```diff
@@ -112,10 +112,7 @@ export class BidiChromium extends BrowserType {
     if (options.devtools)
       chromeArguments.push('--auto-open-devtools-for-tabs');
     if (options.headless) {
-      if (process.env.PLAYWRIGHT_CHROMIUM_USE_HEADLESS_NEW)
-        chromeArguments.push('--headless=new');
-      else
-        chromeArguments.push('--headless=old');
+      chromeArguments.push('--headless=new');
```
Comment (pavelfeldman) [inline packages/playwright-core/src/server/bidi/bidiChromium.ts:115]: is it now simply --headless?

**Review by mxschmitt — APPROVED** [review]
LGTM (after Dimas review)

**Pull request merged**